### PR TITLE
Parse full BitBucket repo urls

### LINKED ISSUE #5874: repository connection parse full url
when adding a repository to an organization, 
`maxbittker/newsdiffs` works but 
`https://bitbucket.org/maxbittker/newsdiffs` doesn't. 
we should recognize and parse this out because several people have run into it

### PR DESCRIPTION
**Pull request by kjlundsgaard**
Fixes #5874

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/utils.jsx [hunk 1]
```diff
@@ -232,6 +234,18 @@ export function getShortVersion(version) {
   return version;
 }
 
+export function parseBitBucketRepo(repo) {
+  if (repo) {
+    let re = /bitbucket\.org\/([^\/]+\/[^\/]+)/i;
+    let match = repo.match(re);
+    let parsedRepo;
+    match ? (parsedRepo = match[1]) : (parsedRepo = repo);
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/utils.jsx]: ik you were just matching the other form, but this should rly be
` let parsedRepo =  match ? match[1] : repo;`
imo

File: src/sentry/static/sentry/app/utils.jsx [hunk 2]
```diff
@@ -232,6 +234,18 @@ export function getShortVersion(version) {
...
+    return parsedRepo;
+  } else {
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/utils.jsx]: also i think it reads easier if the function starts with a guard statement like `if( not repo) return repo`

Comment (MaxBittker) [inline src/sentry/static/sentry/app/utils.jsx]: instead of the big if else that's slightly harder to read

**Review by MaxBittker — CHANGES_REQUESTED** [review]
would it be possible to just have parseRepo, which matches one or the other? would make the code simpler.

also, is there any weird input that could mess with this? like if someone had bitbucket.com as their repo or user name? i guess that's probably not possible. 

I know that this was meant to be a quick fix so feel free to skip some of the style changes if you think it's better not to mess with it right now.

**Review by MaxBittker — APPROVED** [review]
sweeeet :bucket:

**Pull request merged**